### PR TITLE
[convert_to_mixed_precision] fallback to fp32 when encounter circle

### PR DESCRIPTION
**Pull request by jiweibo**
### PR types

Others
### PR changes

Others
### Describe

当遇到环时，涉及算子fallback fp32

### CODE REVIEW

File: paddle/fluid/inference/analysis/passes/convert_to_mixed_precision.cc [hunk 1]
```diff
@@ -411,46 +364,68 @@ void ConvertToMixedPrecisionPass::LoadAndPrepare() {
...
-      for (const auto& name : vars_in_multi_block) {
-        vars_in_multi_block_with_pair_.emplace(
-            name, std::make_pair(VarType::Type(), idx));
-      }
-    }
+  for (auto& name : vars_in_circles) {
+    var_names_in_circles_.insert(name);
+  }
+  for (auto& name : var_names_in_circles_) {
+    LOG(INFO) << name
```
Comment (yuanlehome) [inline paddle/fluid/inference/analysis/passes/convert_to_mixed_precision.cc:394]: LOG(INFO) ?

File: paddle/fluid/inference/analysis/passes/convert_to_mixed_precision.cc [hunk 2]
```diff
@@ -556,94 +502,85 @@ void ConvertToMixedPrecisionPass::ConvertTensorDtype(BlockID block_idx) {
...
+      // If the op has no input or output of float type, we will not choose the
       // low precision kernel.
-      {
-        bool has_float_input{false};
+      if (support_precision) {
+        bool has_float_in_out{false};
         for (auto* in_node : op_node->inputs) {
           if (!in_node->IsVar()) continue;
-          auto* real_node = GetRealVarNode(block_idx, in_node);
+          if (in_node->Var()->GetType() != VarType::LOD_TENSOR) {
```
Comment (yuanlehome) [inline paddle/fluid/inference/analysis/passes/convert_to_mixed_precision.cc:535]: why tensor array input not support?

**Review by Wangzheee — APPROVED** [review]

File: paddle/fluid/inference/analysis/passes/convert_to_mixed_precision.cc (hunk 1 above)
Comment (jiweibo) [reply paddle/fluid/inference/analysis/passes/convert_to_mixed_precision.cc:394]: 是的，显示打印

File: paddle/fluid/inference/analysis/passes/convert_to_mixed_precision.cc (hunk 2 above)
Comment (jiweibo) [reply paddle/fluid/inference/analysis/passes/convert_to_mixed_precision.cc:535]: 不支持tensor array，只支持DenseTensor

**Pull request merged**